src/main/java/org/folio/service/pieces/flows/strategies/ProcessInventoryMixedStrategy.java

```diff
@@ -78,31 +80,39 @@ private void updateLocationWithHoldingInfo(JsonObject holding, Location location
   private void updateLocations(CompositePoLine compPOL) {
     List<Location> locations = new ArrayList<>();
     for (Location location : compPOL.getLocations()) {
-      if (location.getQuantityPhysical() != null && location.getQuantityPhysical() > 0) {
-        Location physicalLocation = new Location();
-        if (PoLineCommonUtil.isHoldingUpdateRequiredForPhysical(compPOL)) {
-          physicalLocation.setHoldingId(location.getHoldingId());
+      boolean physical = location.getQuantityPhysical() != null && location.getQuantityPhysical() > 0;
+      boolean electronic = location.getQuantityElectronic() != null && location.getQuantityElectronic() > 0;
+      boolean physicalUpdate = physical && isHoldingUpdateRequiredForPhysical(compPOL);
+      boolean electronicUpdate = electronic && isHoldingUpdateRequiredForEresource(compPOL);
+      Location newLocation = JsonObject.mapFrom(location).mapTo(Location.class);
+      // Split locations only if one type is getting a holdings update but not the other
+      if (electronicUpdate && physical && !physicalUpdate) {
```

Comment:
maybe there is simpler way, physical && !physicalUpdate - hard to understand

Reply:
I renamed some variables in an attempt to clarify. But the logic is not simple, because the rest of the piece/location handling code has a lot of constraints (I will add info about that in the PR description).

Reply:
I added another comment.